ui/src/virtualconsole/vccuelist.cpp

```diff
@@ -817,25 +817,36 @@ void VCCueList::slotCurrentStepChanged(int stepNumber)
 
         float stepVal;
         int stepsCount = m_tree->topLevelItemCount();
-        if (stepsCount < 256)
-            stepVal = 255.0 / (float)stepsCount;
-        else
+        if (stepsCount < 256) 
+        {
+            stepVal = 256.0 / (float)stepsCount; //divide up the full 0..255 range
+            stepVal = (float)qFloor((stepVal * 100000) + 0.5) / 100000; //round to 5 decimals to fix corner cases
+        } 
+        else 
+        {
             stepVal = 1.0;
-        int slValue = (stepVal * (float)stepNumber);
+        }
+        
+        // value->step# truncates down in slotSideFaderValueChanged; so use ceiling for step#->value
+        float slValue = stepVal * (float)stepNumber;
         if (slValue > 255)
-            slValue = 255;
+            slValue = 255.0;
 
-        int upperBound = 255 - slValue;
-        int lowerBound = qFloor(upperBound - stepVal);
-        //qDebug() << "Slider value:" << m_slider1->value() << "Step range:" << (255 - slValue) << (255 - slValue - stepVal);
+        int upperBound = 255 - qCeil(slValue);
+        int lowerBound = qFloor((float)256.0 - slValue - stepVal);
```

Comment:
I think casting to float is not needed here too since you specified .0 which the compiler interprets as float

Reply:
Tested successfully without the (float). Removed in new commit.